web/src/stream_popover.js

```diff
@@ -597,6 +612,35 @@ export async function build_move_topic_to_stream_popover(
         $("#move_topic_modal .move_messages_edit_topic").on("input", () => {
             update_submit_button_disabled_state(stream_widget_value);
         });
+
+        const default_message_move_count = get_count_of_messages_to_be_moved(
+            args.from_message_actions_popover ? "change_later" : "change_all",
```

Comment:
This can directly be computed from the selected value in the `get_count_of_messages_to_be_moved` function instead of passing this here.

Reply:
Hey @sahil839,

The main purpose behind this approach is that `args.from_message_actions_popover == true` indicates that I clicked on `move messages`, while the `else` case means I clicked on `move topic`. This boolean, `args.from_message_actions_popover`, differentiates between the two options. For `move messages`, the default is to "`move this and the following messages`," and for `move topic`, it should show the total count of messages in the topic.

I’m not sure if it's possible to handle both cases without using this boolean.

An alternative approach would be to introduce a boolean parameter in the `get_count_of_messages_to_be_moved` function.

Could you please advise which approach is better, or if I should consider another option?

Reply:
Ok, yes. I missed that we cannot change propagate mode while opening the modal from topic popover. So, the previous approach was fine I guess.

Reply:
Sure, I reversed the changes.
Thanks!